### ISSUE TITLE
L003: indented_on_contents does not overrule L003

### ISSUE THREAD

**Issue by aloysius-chow** [sqlfluff/sqlfluff#3565]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I noticed that `indented_on_contents = False` is ignored by L003 when linting and reports incorrect indentation.
The following are my configurations and a sample script:
```
[sqlfluff:indentation]
indented_joins = False
indented_using_on = True
indented_on_contents = False

[sqlfluff:rules:L003]
hanging_indents = False
```

```
SELECT
    a.name,
    b.balance
FROM table1 AS a
LEFT JOIN table2 AS b
    ON a.id = b.id
    AND a.country = b.country
```

### Expected Behaviour

There should be no error given.
The documentation at [the indentation page](https://docs.sqlfluff.com/en/stable/indentation.html) states that setting `indented_on_contents` to False will 'align any AND subsections of an ON block with each other', but I can only resolve this by indenting the AND subsection.

### Observed Behaviour

Output:
```
L:   7 | P:   5 | L003 | Expected 2 indentations, found 1 [compared to line 06]
```

### How to reproduce

[Archive.zip](https://github.com/sqlfluff/sqlfluff/files/9059642/Archive.zip)
test.sql and .sqlfluff are included in the zip file.

### Dialect

Redshift

### Version

sqlfluff: 1.0.0
sqlfluff-templater-dbt: 1.0.0
dbt: 1.0.0

### Configuration

```
[sqlfluff]
templater = jinja
dialect = redshift


[sqlfluff:templater:dbt]
project_dir = .
profile = ditto
target = dev


[sqlfluff:indentation]
indented_joins = False
indented_using_on = True
indented_on_contents = False

[sqlfluff:rules]
tab_space_size = 4
indent_unit = space
comma_style = trailing


[sqlfluff:rules:L003]
hanging_indents = False
```
This takes places regardless of whether hanging_indents is set to False or True

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by pdebelak** [comment]
I can replicate this behavior in sqlfluff 1.0.0 but not in sqlfluff 1.1.0. @aloysius-chow can you update and confirm whether this problem is fixed for you in the newest version?

**Comment by tunetheweb** [comment]
This option was only added in 1.1.0:

> Add config for optionally indenting contents of ON blocks https://github.com/sqlfluff/sqlfluff/pull/3471 [@PeterBalsdon](https://github.com/PeterBalsdon)
